src/account_manager.rs

```diff
@@ -20,46 +20,73 @@ use stronghold::Stronghold;
 ///
 /// Used to manage multiple accounts.
 #[derive(Default)]
-pub struct AccountManager {}
+pub struct AccountManager {
+    storage_path: PathBuf,
+}
 
 fn mutate_account_transaction<F: FnOnce(&Account, &mut Vec<Message>)>(
+    storage_path: &PathBuf,
     account_id: AccountIdentifier,
     handler: F,
 ) -> crate::Result<()> {
-    let mut account = crate::storage::get_account(account_id)?;
+    let mut account = crate::storage::get_account(&storage_path, account_id)?;
     let mut transactions: Vec<Message> = account.messages().to_vec();
     handler(&account, &mut transactions);
     account.set_messages(transactions);
-    let adapter = crate::storage::get_adapter()?;
-    adapter.set(account_id, serde_json::to_string(&account)?)?;
+    crate::storage::with_adapter(&storage_path, |storage| {
+        storage.set(account_id, serde_json::to_string(&account)?)
+    })?;
     Ok(())
 }
 
 impl AccountManager {
     /// Initialises a new instance of the account manager with the default storage adapter.
-    pub fn new() -> Self {
-        Default::default()
+    pub fn new() -> crate::Result<Self> {
+        Self::with_storage_path("./example-database")
```

Comment:
If this is temporary, maybe create a constant string would be better.

Reply:
you're right